### PR TITLE
GPDB DOCS - SELECT( INTO) - add WITH clause (common table expression)

### PR DESCRIPTION
**Pull request by mkiyama**
Add WITH clause to the SQL commands
SELECT
SELECT INTO

information from postgres docs and  https://github.com/greenplum-db/gpdb/pull/2394

### CODE REVIEW

File: gpdb-doc/dita/ref_guide/sql_commands/SELECT.xml [hunk 1]
```diff
@@ -625,32 +726,92 @@ CREATE FUNCTION distributors_2(int) RETURNS SETOF record AS
...
+    FROM employee
+    WHERE manager_name = 'Mary'
+  UNION ALL
+    SELECT er.distance + 1, e.employee_name, e.manager_name
+    FROM employee_recursive er, employee e
+    WHERE er.employee_name = e.manager_name
+  )
+SELECT distance, employee_name FROM employee_recursive;
+
+Notice the typical form of recursive queries: an initial condition, followed by UNION, followed by the recursive part of the query. Be sure that the recursive part of the query will eventually return no tuples, or else the query will loop indefinitely. (See Section 7.8 for more examples.)
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/sql_commands/SELECT.xml]: This final paragraph is inside the codeblock.

File: gpdb-doc/dita/ref_guide/sql_commands/SELECT.xml [hunk 2]
```diff
@@ -625,32 +726,92 @@ CREATE FUNCTION distributors_2(int) RETURNS SETOF record AS
...
+</codeblock>
+      <p>The typical form of recursive queries: an initial condition, followed by
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/sql_commands/SELECT.xml]: Change this sentence to start:  "The typical form of a recursive query is:"

File: gpdb-doc/dita/ref_guide/sql_commands/SELECT.xml [hunk 3]
```diff
@@ -1,7 +1,14 @@
...
 <!DOCTYPE topic
   PUBLIC "-//OASIS//DTD DITA Composite//EN" "ditabase.dtd">
-<topic id="topic1"><title id="eg20941">SELECT</title><body><p id="sql_command_desc">Retrieves rows from a table or view.</p><section id="section2"><title>Synopsis</title><codeblock id="sql_command_synopsis">SELECT [ALL | DISTINCT [ON (<varname>expression</varname> [, ...])]]
+<topic id="topic1">
+  <title id="eg20941">SELECT</title>
+  <body>
+    <p id="sql_command_desc">Retrieves rows from a table or view.</p>
+    <section id="section2">
+      <title>Synopsis</title>
+      <codeblock id="sql_command_synopsis">[ WITH [ RECURSIVE ] <varname>with_query</varname> [, ...] ]
```
Comment (hlinnaka) [inline gpdb-doc/dita/ref_guide/sql_commands/SELECT.xml]: RECURSIVE is not supported in GPDB5 yet.

Comment (mkiyama) [inline gpdb-doc/dita/ref_guide/sql_commands/SELECT.xml]: @kavinderd 
I believe Heikki's comment also affects this pull request
https://github.com/greenplum-db/gpdb/pull/2394

**Pull request merged**